### PR TITLE
Remove remaining feature flags

### LINKED ISSUE #242: Target stable Rust
This is a full list of feature flags used by tarpc.

- [external_doc](https://github.com/rust-lang/rust/issues/44732)
- [proc_macro_hygiene](https://github.com/rust-lang/rust/issues/54727)
- [weak_counts](https://github.com/rust-lang/rust/issues/57977)
- [non_exhaustive](https://github.com/rust-lang/rust/issues/44109)
- ~[integer_atomics](https://github.com/rust-lang/rust/issues/32976)~
- ~[try_trait](https://github.com/rust-lang/rust/issues/42327)~
- ~[arbitrary_self_types](https://github.com/rust-lang/rust/issues/44874)~
- [trait_alias](https://github.com/rust-lang/rust/issues/41517)
- [async_await](https://github.com/rust-lang/rust/issues/50547)
- ~[async_closure](https://github.com/rust-lang/rust/issues/62290)~

Now that Futures 1.0 have arrived I believe we can start dropping them one by one until tarpc works on stable Rust.

Continued from #131.

### PR DESCRIPTION
**Pull request by vorot93**
A very hacky solution yet tests pass. Fixes #242.

### CODE REVIEW

**Comment by tikue** [comment]
Left a few minor comments; looks great though! Thank you:)

**Comment by vorot93** [comment]
I have addressed the comments. Please take a look again.

File: rpc/src/server/filter.rs [hunk 1]
```diff
@@ -34,34 +35,49 @@ where
...
     fn drop(&mut self) {
-        // Don't care if the listener is dropped.
-        let _ = self.dropped_keys.unbounded_send(self.key.take().unwrap());
+        if self.counter.count() <= 1 {
+            // Don't care if the listener is dropped.
+            match Arc::try_unwrap(self.key.take().unwrap()) {
+                Ok(key) => {
+                    let _ = self.dropped_keys.unbounded_send(key);
+                }
+                _ => unreachable!(),
```
Comment (tikue) [inline rpc/src/server/filter.rs]: Not: instead of the match, how about calling `Arc::try_unwrap(...).expect("nice error message")`?

Comment (vorot93) [reply rpc/src/server/filter.rs]: That would require K to be Debug

**Pull request merged**